main.go

```diff
@@ -73,6 +79,23 @@ func main() {
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
+	// we have seen in e2e testing that this path can get invoked prior to the Tekton CRD getting generated,
+	// and controller-runtime does not retry on missing CRDs.
+	// so we are going to wait on a Tekton CRDs existing before moving forward.
+	apiextensionsClient := apiextensionsclient.NewForConfigOrDie(ctrl.GetConfigOrDie())
```

Comment:
Could you move this functionality into a separate function?

Reply:
functionality moved to separate function @mmorhun .... PTAL / thanks

Reply:
Thank you, @gabemontero. I think we can merge it.

Reply:
ok cool - do you want to apply the lgtm label via the lgtm prow command, or should we just hit the merge button @mmorhun ?